hazelcast-jet-core/src/test/java/com/hazelcast/jet/pipeline/BatchStageTest.java

```diff
@@ -870,4 +871,19 @@ public void customTransform_keyed() {
         // multiple processors will observe the same keys and the counts won't match.
         assertEquals("0\n1", streamToString(sinkStreamOf(Integer.class), Object::toString));
     }
+
+    @Test
+    public void batchAddTimestampPreserveOrderTest() {
```

Comment:
This can be a unit test I think. It can create a Pipeline, convert it to a DAG and the compare the determined LP of the vertices. It should also have 2 variants: whether the upstream does or does not have a preferred LP in the metaSupplier.